Integrations/EDL/EDL.py

```diff
@@ -0,0 +1,308 @@
+    out_dict[EDL_MIMETYPE_KEY] = 'application/json' if out_format == FORMAT_JSON else 'text/plain'
+    save_context(now, out_dict)
+    return out_dict[EDL_VALUES_KEY]
+
+
+def save_context(now: datetime, out_dict: dict):
+    """Saves EDL state and refresh time to context"""
+    demisto.setLastRun({'last_run': date_to_timestamp(now)})
+    demisto.setIntegrationContext(out_dict)
+
+
+def find_indicators_to_limit(indicator_query: str, limit: int) -> list:
+    """
+    Finds indicators using demisto.findIndicators
+    """
+    iocs, _ = find_indicators_to_limit_loop(indicator_query, limit)
+    return iocs[:limit]
+
+
+def find_indicators_to_limit_loop(indicator_query: str, limit: int, total_fetched: int = 0, next_page: int = 0,
+                                  last_found_len: int = PAGE_SIZE):
+    """
+    Finds indicators using while loop with demisto.findIndicators, and returns result and last page
+    """
+    iocs: List[dict] = []
+    if not last_found_len:
+        last_found_len = total_fetched
+    while last_found_len == PAGE_SIZE and limit and total_fetched < limit:
+        fetched_iocs = demisto.findIndicators(query=indicator_query, page=next_page, size=PAGE_SIZE).get('iocs')
+        iocs.extend(fetched_iocs)
+        last_found_len = len(fetched_iocs)
+        total_fetched += last_found_len
+        next_page += 1
+    return iocs, next_page
+
+
+def create_values_out_dict(iocs: list, out_format: str) -> dict:
+    """
+    Create a dictionary for output values using the selected format (json, json-seq, text, csv)
+    """
+    if out_format == FORMAT_JSON:  # handle json separately
+        iocs_list = [ioc for ioc in iocs]
+        return {EDL_VALUES_KEY: json.dumps(iocs_list)}
+    else:
+        formatted_indicators = []
+        if out_format == FORMAT_CSV and len(iocs) > 0:  # add csv keys as first item
+            headers = list(iocs[0].keys())
+            formatted_indicators.append(list_to_str(headers))
+        for ioc in iocs:
+            value = ioc.get('value')
+            if value:
+                if out_format == FORMAT_TEXT:
+                    formatted_indicators.append(value)
+                elif out_format == FORMAT_JSON_SEQ:
+                    formatted_indicators.append(json.dumps(ioc))
+                elif out_format == FORMAT_CSV:
+                    # wrap csv values with " to escape them
+                    values = list(ioc.values())
+                    formatted_indicators.append(list_to_str(values, map_func=lambda val: f'"{val}"'))
+    return {EDL_VALUES_KEY: list_to_str(formatted_indicators, '\n')}
+
+
+def get_edl_mimetype() -> str:
+    """Returns the mimetype of the EDL"""
+    ctx = demisto.getIntegrationContext()
+    return ctx.get(EDL_MIMETYPE_KEY, 'text/plain')
+
+
+def get_edl_ioc_values() -> str:
+    """
+    Get the ioc list to return in the edl
+    """
+    params = demisto.params()
+    out_format = params.get('format')
```

Comment:
pass these arguments to `get_edl_ioc_values` instead of using global object demisto